src/crimson/os/seastore/journal/circular_bounded_journal.h

```diff
@@ -179,6 +180,38 @@ class CircularBoundedJournal : public Journal {
 
   submit_record_ret do_submit_record(record_t &&record, OrderingHandle &handle);
 
+  void try_read_rolled_header(scan_valid_records_cursor &cursor) {
+    paddr_t addr = convert_abs_addr_to_paddr(
+      get_records_start(),
+      get_device_id());
+    cursor.seq.offset = addr;
+    cursor.seq.segment_seq += 1;
+  }
+
+  void initialize_cursor(scan_valid_records_cursor& cursor) final {
+    cursor.block_size = get_block_size();
+  };
+
+  Journal::replay_ret replay_segment(
+    cbj_delta_handler_t &handler, scan_valid_records_cursor& cursor);
+
+  read_ret read(paddr_t start, size_t len) final;
+
+  bool is_passed_end(scan_valid_records_cursor &cursor,
+    record_group_header_t &h) final;
+
+  int64_t get_segment_end_offset(paddr_t addr) final {
+    return get_journal_end();
+  }
+
+  void update_committed_to(journal_seq_t seq) {
```

Comment:
Doesn't look necessary anymore, revert to the plain `record_submitter.update_committed_to(get_written_to())`.

Reply:
Done.